File: solutions/C++/number_of_1_bits_loop.cpp
#include <cstdint>
using namespace std;

int numberOf1BitsLoop(uint32_t n) {
    int count = 0;
    while (n) {
        count += n & 1;
        n >>= 1;
    }
    return count;
}

int main() {
    cout << numberOf1BitsLoop(11) << endl;  // 3
    return 0;
}

File: solutions/JavaScript/number_of_1_bits_loop.js
function numberOf1BitsLoop(n) {
    let count = 0;
    while (n) {
        count += n & 1;
        n >>>= 1;
    }
    return count;
}

console.log(numberOf1BitsLoop(11));  // 3
console.log(numberOf1BitsLoop(128));  // 1
